src/ast.mli

```diff
@@ -0,0 +1,91 @@
+type program = None
+
+type bin_op =
+  Add    | Minus | Multiply | Divide
+  | Modulo | Equal | Neq      | Lt
+  | Leq    | Gt    | Geq      | And
+  | Or
+
+type u_op = Not | Neg
+
+type p_type = Int_t | Bool_t | Double_t | Char_t | Unit_t
+
+type cont_type = String_t | List_t | Set_t | Map_t | Tuple_t
+
+type actor_op =
+  | Actor_send
+  | Actor_broadcast
+  | Actor_receive
+
+type types =
```

Comment:
Actually I'm talking nonsense, for some reason I was assuming that for this particular case the ast would automatically know that only None_param should be selected if the type is none. This is a quick fix though.